adapters/adapters-test-utils/src/main/java/uk/gov/justice/services/adapters/test/utils/builder/HttpActionBuilder.java

```diff
@@ -0,0 +1,174 @@
+import org.raml.model.Response;
+import org.raml.model.parameter.QueryParameter;
+
+/**
+ * Builds RAML http action (not to be confused with framework's action)
+ */
+public class HttpActionBuilder {
+
+    private final Map<String, MimeType> body = new HashMap<>();
+    private final Map<String, QueryParameter> queryParameters = new HashMap<>();
+    private ActionType actionType;
+    private List<Response> responses = new LinkedList<>();
+    private String description;
+    private MappingDescriptionBuilder mappingDescription;
+
+    public static HttpActionBuilder httpAction() {
+        return new HttpActionBuilder();
+    }
+
+    public static HttpActionBuilder defaultPostAction() {
+        return httpAction()
+                .withHttpActionType(POST)
+                .withHttpActionOfDefaultRequestType();
+    }
+
+    public static HttpActionBuilder defaultGetAction() {
+        return httpAction()
+                .withHttpActionType(GET)
+                .withDefaultResponseType();
+    }
+
+    public static HttpActionBuilder httpAction(final ActionType actionType, final String... mimeTypes) {
+        HttpActionBuilder httpActionBuilder = new HttpActionBuilder()
+                .withHttpActionType(actionType)
+                .with(defaultMapping());
+        for (String mimeType : mimeTypes) {
+            httpActionBuilder = httpActionBuilder.withMediaType(mimeType);
+        }
+        return httpActionBuilder;
+    }
+
+    public static HttpActionBuilder httpActionWithNoMapping(final ActionType actionType, final String... mimeTypes) {
+        HttpActionBuilder httpActionBuilder = new HttpActionBuilder()
+                .withHttpActionType(actionType);
+        for (String mimeType : mimeTypes) {
+            httpActionBuilder = httpActionBuilder.withMediaType(mimeType);
+        }
+        return httpActionBuilder;
+    }
+
+    public HttpActionBuilder withHttpActionType(final ActionType actionType) {
+        this.actionType = actionType;
+        return this;
+    }
+
+    public HttpActionBuilder withHttpActionOfDefaultRequestType() {
+        return withMediaType("application/vnd.ctx.command.defcmd+json")
+                .with(mapping()
+                        .withName("action1")
+                        .withRequestType("application/vnd.ctx.command.defcmd+json"));
+    }
+
+    public HttpActionBuilder withDefaultResponseType() {
+        return withResponseTypes("application/vnd.ctx.query.defquery+json")
+                .with(mapping()
+                        .withName("action1")
+                        .withResponseType("application/vnd.ctx.query.defquery+json"));
+    }
+
+    public HttpActionBuilder withResponseTypes(final String... responseTypes) {
+        return withHttpActionResponse(new Response(), responseTypes);
+    }
+
+    public HttpActionBuilder withHttpActionResponse(final Response response, final String... responseTypes) {
+        Map<String, MimeType> respBody = new HashMap<>();
```

Comment:
There are final's missing here and in for loops etc (above too), but not sure what the policy is if we need them in test-utils.